assets/JS/app.js

```diff
@@ -0,0 +1,101 @@
+// Get the form and table body elements from the DOM
+var productForm = document.getElementById("productForm"); // Select the form
+var cartTableBody = document.querySelector("#cartTable tbody"); // Select the table body
+
+// Function to save the table data to Local Storage
+function saveToLocalStorage() {
+    // Create an empty array to store rows
+    var rows = [];
+    
+    // Get all the rows in the table
+    var tableRows = document.querySelectorAll("#cartTable tbody tr");
+    for (var i = 0; i < tableRows.length; i++) {
+        var row = tableRows[i];
+        
+        // Create an object with the row data
+        var product = {
+            name: row.cells[0].textContent, // First cell: product name
+            price: row.cells[1].textContent, // Second cell: product price
+            category: row.cells[2].textContent, // Third cell: product category
+            image: row.cells[3].querySelector("img").src // Fourth cell: product image URL
+        };
+        
+        // Add the product object to the array
+        rows.push(product);
+    }
+    
+    // Save the array to Local Storage as a string
+    localStorage.setItem("cartData", JSON.stringify(rows));
+}
+
+// Function to load data from Local Storage
+function loadFromLocalStorage() {
+    // Get saved data from Local Storage
+    var savedData = localStorage.getItem("cartData");
+    
+    // Check if there is data
+    if (savedData) {
+        // Convert the JSON string back to an array
+        var rows = JSON.parse(savedData);
+        
+        // Loop through each product and add it to the table
+        for (var i = 0; i < rows.length; i++) {
+            var product = rows[i];
+            addRowToTable(product.name, product.price, product.category, product.image);
```

Comment:
you're manipulating the DOM in a loop which is a bad practice. the browser needs to re-render any iteration. a better practice is to cumulate changes in a variable and then issue a single DOM manipulation